Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/survey/SurveyModule.kt

```diff
@@ -1,9 +1,62 @@
 package de.rki.coronawarnapp.datadonation.survey
 
+import android.content.Context
 import dagger.Module
+import dagger.Provides
+import dagger.Reusable
 import de.rki.coronawarnapp.datadonation.survey.consent.SurveyConsentModule
+import de.rki.coronawarnapp.datadonation.survey.server.DataDonationApiV1
+import de.rki.coronawarnapp.datadonation.survey.server.DataDonationHttpClient
+import de.rki.coronawarnapp.environment.datadonation.DataDonationCDNServerUrl
+import de.rki.coronawarnapp.http.HttpClientDefault
+import de.rki.coronawarnapp.http.RestrictedConnectionSpecs
+import de.rki.coronawarnapp.util.di.AppContext
+import okhttp3.Cache
+import okhttp3.ConnectionSpec
+import okhttp3.OkHttpClient
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import retrofit2.converter.protobuf.ProtoConverterFactory
+import java.io.File
+import javax.inject.Singleton
 
 @Module(
     includes = [SurveyConsentModule::class]
 )
-class SurveyModule
+class SurveyModule {
+
+    @Reusable
+    @DataDonationHttpClient
+    @Provides
+    fun cdnHttpClient(
+        @HttpClientDefault defaultHttpClient: OkHttpClient,
+        @RestrictedConnectionSpecs connectionSpecs: List<ConnectionSpec>
+    ): OkHttpClient =
+        defaultHttpClient.newBuilder().connectionSpecs(connectionSpecs).build()
+
+    @Singleton
+    @Provides
+    fun provideDataDonationApi(
+        @AppContext context: Context,
+        @DataDonationHttpClient client: OkHttpClient,
+        @DataDonationCDNServerUrl url: String,
+        protoConverterFactory: ProtoConverterFactory,
+        gsonConverterFactory: GsonConverterFactory
+    ): DataDonationApiV1 {
+        val cache = Cache(File(context.cacheDir, "http_data_donation"), DEFAULT_CACHE_SIZE)
```

Comment:
Please keep in mind that this is the module for `Survey` related http :mage: only.
Does the survey need http request caching, and if it does, what would be affected, are there any GET requests that use the cache?
Would POST requests use the cache too, and could this create an issue where we receive a cached OTP token instead of a new one?

Reply:
removed the cache